libraries/chain/abi_serializer.cpp

```diff
@@ -186,24 +187,22 @@ namespace eosio { namespace chain {
       return ends_with(string(type), "[]");
    }
 
-   // 1-dimensonal fixed size array
+   // one-dimensional fixed size array
    bool _is_fixed_array(const type_name& type, int &typelen, int &arrsize) {
-      string stype = type;
-      int p = 1;
       typelen = 0;
       arrsize = 0;
-      while (p < stype.length() && stype[p] != '[') ++p;
-      if (p >= stype.length()) return false;
+      auto p = type.find_first_of('[');
+      if (p == string::npos || p == 0) return false;
       typelen = p;
       ++p;
-      if (p >= stype.length()) return false;
-      while (p < stype.length() && (stype[p] >= '0' && stype[p] <= '9')) {
-         arrsize = arrsize * 10 + (stype[p] - '0');
+      if (p >= type.length()) return false;
+      while (p < type.length() && (type[p] >= '0' && type[p] <= '9')) {
+         arrsize = arrsize * 10 + (type[p] - '0');
          if (arrsize > (size_t)MAX_NUM_ARRAY_ELEMENTS) return false;
          ++p;
       }
       if (!arrsize) return false;
-      if (p != stype.length() - 1 || stype[p] != ']') return false;
+      if (p != type.length() - 1 || type[p] != ']') return false;
```

Comment:
I think we should find_first_of(']') and set its own variable, at same time as setting variable p, and then add that to the if check, so we don't waste our time parsing typelen and arrsize when the fixed array definition is invalid and the code will be cleaner
auto p = type.find_first_of('[');
const auto q = type.find_first_of(']');
if (p == npos || q== npos || p == 0 || (p - q < 2)) return false;